lib/rpmsg/rpmsg.c

```diff
@@ -246,6 +263,7 @@ static void rpmsg_unregister_endpoint(struct rpmsg_endpoint *ept)
 				      ept->addr);
 	metal_list_del(&ept->node);
 	ept->rdev = NULL;
+	rpmsg_ept_decref(ept);
```

Comment:
Here we can have an issue if `ept->rdev` is used in `ept->release_cb`
move `ept->rdev = NULL` in rpmsg_ept_decref() should avoid issue

```
void rpmsg_ept_decref(struct rpmsg_endpoint *ept)
{
	if (ept && ept->release_cb) {
		ept->refcnt--;
		if (!ept->refcnt) 
			ept->release_cb(ept);
		else 
			ept->rdev = NULL;
	}
}
```

Reply:
thanks, but I think there may be issues with such changes. firstly, not all endpoints require registration for release_cb; Secondly, the first call to rpmsg_ept_cb should be after rpmsg_xxx_ept_cb， but at that time, refcnt will not decrease to 0, setting ept ->rdev=NULL directly can also cause problems.

Reply:
after release_cb, it's invalid to touch any fields inside ept.

Reply:
oops sorry my proposal contains a bug....
The idea is to either call the` release_cb()` or to set `ept->rdev` to `NULL`

```
void rpmsg_ept_incref(struct rpmsg_endpoint *ept)
{
	if (ept) {
		ept->refcnt++;
	}
}

void rpmsg_ept_decref(struct rpmsg_endpoint *ept)
{
	if (ept) {
		ept->refcnt--;
		if (!ept->refcnt) 
			if (ept->release_cb)
				ept->release_cb(ept);
			else 
				ept->rdev = NULL;
	}
}
```